src/satellite_images_nso/_manipulation/nso_manipulator.py

```diff
@@ -155,6 +156,7 @@ def add_index_channels(tif_input_file: str, channel_types: list):
             output_dataset.write_band(dataset.count + 1 + i, index_channels[i])
         output_dataset.descriptions = descriptions
 
+    os.remove(tif_input_file)

```

Comment:
if we add a channel to the original .tif the original tif file would not be deleted. I think this will lead to unnecessary data duplication. Thus i will delete the original .tif file after the channel has been added

Reply:
I'm not opposed to deleting the initial tif file, I'm just opposed to mixing functionality (file manipulation and data processing) in a single function. But let's leave it for now.